ignore_test.go

```diff
@@ -116,170 +117,205 @@ func TestCompileIgnoreLines_HandleSpaces(test *testing.T) {
+	assert.Equal(test, true, object.MatchesPath("external/barfoo/.bower.json"), "external/barfoo/.bower.json")
 }
 
 func TestCompileIgnoreLines_CarriageReturn(test *testing.T) {
-    lines := []string{"abc/def\r", "a/b/c\r", "b\r"}
-    object, error := CompileIgnoreLines(lines...)
-    assert.Nil(test, error, "error from CompileIgnoreLines should be nil")
+	lines := []string{"abc/def\r", "a/b/c\r", "b\r"}
+	object, error := CompileIgnoreLines(lines...)
+	assert.Nil(test, error, "error from CompileIgnoreLines should be nil")
 
-    assert.Equal(test, true,  object.MatchesPath("abc/def/child"), "abc/def/child should match")
-    assert.Equal(test, true,  object.MatchesPath("a/b/c/d"),       "a/b/c/d should match")
+	assert.Equal(test, true, object.MatchesPath("abc/def/child"), "abc/def/child should match")
+	assert.Equal(test, true, object.MatchesPath("a/b/c/d"), "a/b/c/d should match")
 
-    assert.Equal(test, false, object.MatchesPath("abc"), "abc should not match")
-    assert.Equal(test, false, object.MatchesPath("def"), "def should not match")
-    assert.Equal(test, false, object.MatchesPath("bd"),  "bd should not match")
+	assert.Equal(test, false, object.MatchesPath("abc"), "abc should not match")
+	assert.Equal(test, false, object.MatchesPath("def"), "def should not match")
+	assert.Equal(test, false, object.MatchesPath("bd"), "bd should not match")
 }
 
 func TestCompileIgnoreLines_WindowsPath(test *testing.T) {
-    if runtime.GOOS != "windows" {
-        return
-    }
-    lines := []string{"abc/def", "a/b/c", "b"}
-    object, error := CompileIgnoreLines(lines...)
-    assert.Nil(test, error, "error from CompileIgnoreLines should be nil")
-
-    assert.Equal(test, true,  object.MatchesPath("abc\\def\\child"), "abc\\def\\child should match")
-    assert.Equal(test, true,  object.MatchesPath("a\\b\\c\\d"),       "a\\b\\c\\d should match")
-}
\ No newline at end of file
+	if runtime.GOOS != "windows" {
+		return
+	}
+	lines := []string{"abc/def", "a/b/c", "b"}
+	object, error := CompileIgnoreLines(lines...)
+	assert.Nil(test, error, "error from CompileIgnoreLines should be nil")
+
+	assert.Equal(test, true, object.MatchesPath("abc\\def\\child"), "abc\\def\\child should match")
+	assert.Equal(test, true, object.MatchesPath("a\\b\\c\\d"), "a\\b\\c\\d should match")
+}
+
+func TestWildCardFiles(test *testing.T) {
+	gitIgnore := []string{"*.swp", "/foo/*.wat", "bar/*.txt"}
+	object, err := CompileIgnoreLines(gitIgnore...)
+	assert.Nil(test, err, "error from CompileIgnoreLines should be nil")
+
+	// Paths which are targeted by the above "lines"
+	assert.Equal(test, true, object.MatchesPath("yo.swp"), "should ignore all swp files")
+	assert.Equal(test, true, object.MatchesPath("something/else/but/it/hasyo.swp"), "should ignore all swp files in other directories")
+
+	assert.Equal(test, true, object.MatchesPath("foo/bar.wat"), "should ignore all wat files in foo - nonpreceding /")
```

Comment:
from line 297 to 301 i describe the test case i mentioned in the PR where
you have `/foo/*.bar` in your .gitignore` and `bar/*.txt`